packages/graph/hash_graph/lib/graph/src/shared/subgraph/edges/mod.rs

```diff
@@ -1,175 +1,130 @@
 use std::collections::{hash_map::Entry, HashMap, HashSet};
 
-use serde::Serialize;
-use type_system::uri::BaseUri;
-use utoipa::{openapi, ToSchema};
+use serde::{Serialize, Serializer};
 
-use crate::identifier::{
-    knowledge::{EntityId, EntityVersion},
-    ontology::OntologyTypeVersion,
-    GraphElementEditionId,
-};
+use crate::identifier::{knowledge::EntityEditionId, ontology::OntologyTypeEditionId};
 
 mod edge;
 mod kind;
 
 pub use self::{
-    edge::{GenericOutwardEdge, KnowledgeGraphOutwardEdges, OntologyOutwardEdges, OutwardEdge},
+    edge::{KnowledgeGraphOutwardEdges, OntologyOutwardEdges, OutwardEdge},
     kind::{KnowledgeGraphEdgeKind, OntologyEdgeKind, SharedEdgeKind},
 };
 
-#[derive(Default, Debug, Serialize)]
-#[serde(transparent)]
-pub struct OntologyRootedEdges(
-    // TODO: expose it through methods instead of making this field `pub`
-    //   see https://app.asana.com/0/1202805690238892/1203358665695491/f
-    pub HashMap<BaseUri, HashMap<OntologyTypeVersion, HashSet<OntologyOutwardEdges>>>,
-);
-
-// This is needed because Utoipa doesn't know how to handle the BTreeMaps
-impl ToSchema for OntologyRootedEdges {
-    fn schema() -> openapi::Schema {
-        openapi::ObjectBuilder::new()
-            .additional_properties(Some(openapi::Schema::from(
-                openapi::ObjectBuilder::new().additional_properties(Some(openapi::Array::new(
-                    openapi::Ref::from_schema_name("OntologyOutwardEdges"),
-                ))),
-            )))
-            .into()
-    }
+#[derive(Default, Debug)]
+pub struct Edges {
+    ontology: HashMap<OntologyTypeEditionId, HashSet<OntologyOutwardEdges>>,
```

Comment:
Flattening it like this means we can't easily swap to a `BTreeMap` for the inner map which we did want to do. We could still do that in Utoipa but I worry that the cost is going to be higher now rather than inserting it in a sorted fashion from the start.

Reply:
(Not asking for a change, just commenting)

Reply:
We can use a BTreeMap in the utoipa-representation.

Reply:
> but I worry that the cost is going to be higher now rather than inserting it in a sorted fashion from the start.